src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AwaitExpressionSpiller.cs

```diff
@@ -372,34 +372,42 @@ private BoundStatement UpdateStatement(BoundSpillSequenceBuilder builder, BoundS
 
                     case BoundKind.FieldAccess:
                         var field = (BoundFieldAccess)expression;
-                        if (field.FieldSymbol.IsReadOnly)
+                        var fiedlSymbol = field.FieldSymbol;
+                        if (fiedlSymbol.IsStatic)
                         {
-                            if (field.FieldSymbol.IsStatic) return field;
-                            if (field.FieldSymbol.ContainingType.IsValueType) goto default;
-                            // save the receiver; can get the field later.
-                            var receiver = Spill(builder, field.ReceiverOpt, (refKind != RefKind.None && field.FieldSymbol.Type.IsReferenceType) ? refKind : RefKind.None, sideEffectsOnly);
-                            return field.Update(receiver, field.FieldSymbol, field.ConstantValueOpt, field.ResultKind, field.Type);
+                            // no need to spill static fields if used as locations or if readonly
+                            if (refKind != RefKind.None || fiedlSymbol.IsReadOnly)
+                            {
+                                return field;
+                            }
+                            goto default;
                         }
-                        goto default;
+
+                        if (refKind == RefKind.None) goto default;
+
+                        var receiver = Spill(builder, field.ReceiverOpt, field.FieldSymbol.ContainingType.IsValueType ? refKind : RefKind.None);
+                        return field.Update(receiver, field.FieldSymbol, field.ConstantValueOpt, field.ResultKind, field.Type);
 
                     case BoundKind.Call:
```

Comment:
By the time we get to stack spilling, there should be only method calls, no properties/events and such. All should be lowered to calls.